disnake/state.py

```diff
@@ -280,6 +281,7 @@ def clear(self, *, views: bool = True, application_commands: bool = True) -> Non
         self._emojis: Dict[int, Emoji] = {}
         self._stickers: Dict[int, GuildSticker] = {}
         self._guilds: Dict[int, Guild] = {}
+        self._scheduled_event_guilds: Dict[int, int] = {}
```

Comment:
This is pretty funky. Why can't we do the same as we do for channels, etc.?

Reply:
> Why can't we do the same as we do for channels, etc.?
You mean looping through all guilds and calling `Guil.get_scheduled_event` until it's not `None`? This is too slow for an operation that can potentially be called multiple times per second. I should probably mention that this solution is temporal due to the comment above.